src/index.js

```diff
@@ -17,6 +17,20 @@ module.exports = (config = {}) => {
     kinesis.endpoint = config.endpoint;
   }
 
+  if (config.type && config.type === 'BATCH') {
+    return (events) => {
+
+      if (!Array.isArray(events)) throw 'Events needs to be an Array.';
+
+      if (!events.length) throw 'Events are missing.';
+
+      // Each PutRecords request can support up to only 500 records.
+      if (events.length > 500) throw 'Events array can only have 500 records.';
```

Comment:
same here, instead of throwing an exception, we could instead partition the events here in groups of 500, and then emit each partition separately, i.e. in pseudo code
```
return events.in_batch_of(500).map { |batch|
  emitEvents(...)
}
```
what do you think?

Reply:
again, the batching should be left to consumers honestly.
each project's batching needs might differ from the next one, but each batch cannot have more than 500 events, so we just throw back the appropriate error and the consumers can decide to go for a smaller batch and the order of emitting as well.

Reply:
and again, as a consumer of the library I don't need to know that max is 500 items, I would care if I would use AWS SDK directly, as a user if I'm using abstraction like miza-kinesis, I want it to cover such details for me

Reply:
yeah, makes sense. I updated the code accordingly.